core/sdk-core/src/main/java/software/amazon/awssdk/core/internal/useragent/IdentityProviderNameMapping.java

```diff
@@ -0,0 +1,88 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+package software.amazon.awssdk.core.internal.useragent;
+
+import java.util.Locale;
+import java.util.Map;
+import java.util.Optional;
+import java.util.regex.Pattern;
+import software.amazon.awssdk.utils.internal.EnumUtils;
+
+/**
+ * A enum class representing a short form of identity providers to record in the UA string.
+ */
+public enum IdentityProviderNameMapping {
+
+    SYS("SystemPropertyCredentialsProvider"),
+    ENV("EnvironmentVariableCredentialsProvider"),
+    STSWEB("StsAssumeRoleWithWebIdentity"),
+    STSROLE("StsAssumeRoleCredentialsProvider"),
+    STSSAML("StsAssumeRoleWithWebIdentityCredentialsProvider"),
+    STSFED("StsGetFederationTokenCredentialsProvider"),
+    STSSESS("StsGetSessionTokenCredentialsProvider"),
+    SSO("SsoCredentialsProvider"),
+    PROF("ProfileCredentialsProvider"),
+    CONT("ContainerCredentialsProvider"),
+    IMDS("InstanceProfileCredentialsProvider"),
+    STAT("StaticCredentialsProvider"),
+    PROC("ProcessCredentialsProvider"),
+    ANON("AnonymousCredentialsProvider"),
```

Comment:
How do we ensure that in future when we add new  AwsCredentialsProviders  in SDK gets added to this map ? Is there way we can automate ail fast test if we miss this ?

Reply:
That's a good reflection. I thought about as well - it IS brittle. I think in this instance, it's ok, because I expect this will be a relatively short-lived piece of code with the new ways of tracking features in user agent that are coming soon and will change / invalidate this class.

Reply:
okay , got it 👍